ui/src/app/components/access-renewal-notification.tsx

```diff
@@ -0,0 +1,108 @@
+import {Button} from './buttons';
+import {FlexRow} from './flex';
+import {AlarmExclamation} from './icons';
+
+const styles = reactStyles({
+  box: {
+    boxSizing: 'border-box',
+    height: '56.5px',
+    width: '380.5px',
+    border: '0.5px solid rgba(38,34,98,0.5)',
+    borderRadius: '5px',
+    backgroundColor: 'rgba(38,34,98,0.08)',
+    alignItems: 'center',
+    justifyContent: 'center',
+    marginLeft: 'auto',
+  },
+  icon: {
+    height: '25px',
+    width: '25px',
+    color:  colors.warning,
+    fontFamily: 'Font Awesome 5 Pro',
+    fontSize: '25px',
+    letterSpacing: 0,
+    lineHeight: '25px',
+    marginLeft: 'auto',
+  },
+  text: {
+    height: '40px',
+    width: '177px',
+    color: colors.primary,
+    fontFamily: 'Montserrat',
+    fontSize: '14px',
+    fontWeight: 600,
+    letterSpacing: 0,
+    lineHeight: '20px',
+    marginLeft: 'auto',
+  },
+  button: {
+    boxSizing: 'border-box',
+    height: '30px',
+    width: '102px',
+    border: '0.8px solid #216FB4',
+    borderRadius: '1.6px',
+    marginLeft: 'auto',
+    marginRight: 'auto',
+  },
+  buttonText: {
+    height: '20px',
+    width: '79.6px',
+    fontFamily: 'Montserrat',
+    fontSize: '11.2px',
+    fontWeight: 500,
+    letterSpacing: '-0.24px',
+    lineHeight: '19.2px',
+    textAlign: 'center',
+  },
+});
+
+export const MILLIS_PER_DAY = 24 * 60 * 60 * 1000;
+export const NOTIFICATION_THRESHOLD_DAYS = 30;
+
+// return the number of full days remaining to expiration in the soonest-to-expire module,
+// but only if it is within the threshold.
+// if it is not, or no expiration dates are present in the profile: return undefined.
+export const maybeDaysRemaining = (profile: Profile): number | undefined => {
+  const earliestExpiration: number = fp.flow(
+    fp.get(['renewableAccessModules', 'modules']),
+    fp.map<RenewableAccessModuleStatus, number>(m => m.expirationEpochMillis),
+    // remove the undefined expirationEpochMillis
+    fp.compact,
+    fp.min)(profile);
+
+  if (earliestExpiration) {
+    const daysRemaining = (earliestExpiration - Date.now()) / MILLIS_PER_DAY;
+    if (daysRemaining < NOTIFICATION_THRESHOLD_DAYS) {
+      // note that we will show 0 days remaining if the expiration is later today
+      return Math.trunc(daysRemaining);
+    }
+  }
+};
+
+const AccessRenewalNotification = (props: {daysRemaining: number}) => {
+  return <FlexRow style={styles.box}>
+    <AlarmExclamation style={styles.icon}/>
+    <div style={styles.text}>Time for access renewal. {props.daysRemaining} days remaining.</div>
+    {/* TODO RW-6618 navigate to Access Renewal pages */}
+    <Button type='primary' style={styles.button} onClick={() => navigateByUrl('profile')}>
+      <div style={styles.buttonText}>Get Started</div>
+    </Button>
+  </FlexRow>;
+};
+
+export const AccessRenewalNotificationMaybe = (props: {profile: Profile}) => {
+  const daysRemaining = maybeDaysRemaining(props.profile);
+  if (!!daysRemaining) {
```

Comment:
nit: You could use a ternary
```javascript
// null is a valid return value from a render function in react
return !!daysRemaining ? <AccessRenewalNotification daysRemaining={daysRemaining}/> : null
```

I agree with your comment below. Having some comments like this in the code, even if it is repetitive, can be helpful to those unfamiliar with the code base / coding style or new to UI, unfamiliar with the framework, backend, or w/e programming